C++ API parity: Hardshrink

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27060 C++ API parity: LogSigmoid
* #27059 C++ API parity: LeakyReLU
* #27038 C++ API parity: Hardtanh
* **#27035 C++ API parity: Hardshrink**

Differential Revision: [D17682403](https://our.internmc.facebook.com/intern/diff/D17682403)

pbelevich committed:
C++ API parity: Hardshrink

[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -16,5 +16,15 @@ struct ELUOptions {
+struct HardshrinkOptions {
+  HardshrinkOptions() {}
+
+  // the lambda value for the Hardshrink formulation. Default: 0.5
```
nit: we need `///` instead of `//` in order for the comment to show up in the docs

yf225 commented on torch/csrc/api/src/nn/modules/activation.cpp:
```diff
@@ -20,5 +20,21 @@ void ELUImpl::pretty_print(std::ostream& stream) const {
+
+void HardshrinkImpl::pretty_print(std::ostream& stream) const {
+  stream << std::boolalpha
+         << "torch::nn::Hardshrink(lambda=" << options.lambda() << ")";
```
It seems that the Python version only prints `Hardshrink(0.5)` instead of `Hardshrink(lambda=0.5)`, I think we can consider matching the Python version here.

pbelevich committed:
Update on "C++ API parity: Hardshrink"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Hardshrink"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: Hardshrink"


Differential Revision: [D17682403](https://our.internmc.facebook.com/intern/diff/D17682403)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Hardshrink"


Differential Revision: [D17682403](https://our.internmc.facebook.com/intern/diff/D17682403)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Hardshrink"


Differential Revision: [D17682403](https://our.internmc.facebook.com/intern/diff/D17682403)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Hardshrink"


Differential Revision: [D17682403](https://our.internmc.facebook.com/intern/diff/D17682403)

[ghstack-poisoned]
